private-templates-service/adaptivecards-templating-service/src/storageproviders/MongoDBProvider.ts

```diff
@@ -15,23 +18,28 @@ export class MongoDBProvider implements StorageProvider {
   // provided in the query, then we need to add operator $all
   // to make sure they're both in the returned user 'team' field
   private _constructUserQuery(query: Partial<IUser>): any {
-    let userQuery: any = { ...query };
-    if (query.team) {
+    let userQuery: any = MongoUtils.removeUndefinedFields(query);
+    if (query.team && query.team.length) {
       userQuery.team = { $all: query.team };
     }
-    if (query.org) {
+    if (query.org && query.org.length) {
       userQuery.org = { $all: query.org };
     }
     return userQuery;
   }
   private _constructTemplateQuery(query: Partial<ITemplate>): any {
-    let templateQuery: any = { ...query };
+    let templateQuery: any = MongoUtils.removeUndefinedFields(query);
     if (query.name) {
       templateQuery.name = { $regex: query.name, $options: "i" };
     }
-    if (query.tags) {
-      templateQuery.tags = { $all: query.tags };
+    if (query.tags && query.tags.length) {
+      templateQuery.tags = {
+        $all: clone(query.tags).map(x => {
+          return x.toLocaleLowerCase();
```

Comment:
also maybe handle removing whitespace (.trim) if that hasn't been done already.

Reply:
I think it's gonna be done elsewhere before passing it here, I just do here because all tags must be lowercase :)